addons/missileguidance/functions/fnc_doppler_onFired.sqf

```diff
@@ -0,0 +1,83 @@
+#include "..\script_component.hpp"
+/*
+ * Author: tcvm
+ * Sets up doppler radar state arrays (called from missileGuidance's onFired).
+ *
+ * Arguments:
+ * Guidance Arg Array <ARRAY>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [] call ace_missileguidance_fnc_mwr_onFired
+ *
+ * Public: No
+ */
+params ["_firedEH", "_launchParams", "", "", "_stateParams"];
+_firedEH params ["_shooter","","","","","","_projectile"];
+_stateParams params ["", "_seekerStateParams"];
+_launchParams params ["","_targetLaunchParams"];
+_targetLaunchParams params ["_target"];
+
+if (isNull _target && isVehicleRadarOn vehicle _shooter) then {
+    _target = cursorTarget;
+};
+// always allow tracking of projectiles
+if !(_target isKindOf "AllVehicles" || { (typeOf _target) isKindOf ["Default", configFile >> "CfgAmmo"] }) then {
+    _target = nil;
+};
+_launchParams set [0, _target];
+_projectile setMissileTarget objNull; // to emulate a no launch warning
+
+private _projectileConfig = configOf _projectile;
+private _config = _projectileConfig >> "ace_missileguidance";
+
+private _isActive = false;
+private _activeRadarDistance = [_config >> "activeRadarEngageDistance", "NUMBER", 500] call CBA_fnc_getConfigEntry;
+private _projectileThrust = [_projectileConfig >> "thrust", "NUMBER", 0] call CBA_fnc_getConfigEntry;
+private _projectileThrustTime = [_projectileConfig >> "thrustTime", "NUMBER", 0] call CBA_fnc_getConfigEntry;
```

Comment:
If the default is the default that is gotten with `getX`, then use `getX` directly. `CBA_fnc_getConfigEntry` is about 2-3x more expensive in terms of performance, so imo only use when needed.

```suggestion
private _projectileThrust = getNumber (_projectileConfig >> "thrust");
private _projectileThrustTime = getNumber (_projectileConfig >> "thrustTime");
```